com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -1389,41 +1387,42 @@ private bool ExportInstance(GameObject unityGo, FbxScene fbxScene, FbxNode fbxNo
                 return false;
             }
 
-            Object unityPrefabParent = PrefabUtility.GetCorrespondingObjectFromSource(unityGo);
-
+            // where the fbx mesh is stored on a successful export
             FbxMesh fbxMesh = null;
+            // store the shared mesh of the game object
+            Mesh unityGoMesh = null;
+            
+            // get the mesh of the game object
+            if (unityGo.TryGetComponent<MeshFilter>(out MeshFilter meshFilter))
+            {
+                unityGoMesh = meshFilter.sharedMesh;
+            }
 
-            if (unityPrefabParent != null && !SharedMeshes.TryGetValue (unityPrefabParent.GetInstanceID(), out fbxMesh))
+            if (!unityGoMesh)
             {
-                if (Verbose)
-                    Debug.Log (string.Format ("exporting instance {0}({1})", unityGo.name, unityPrefabParent.name));
-                
-                if (ExportMesh (unityGo, fbxNode) && fbxNode.GetMesh() != null) {
-                    SharedMeshes [unityPrefabParent.GetInstanceID()] = fbxNode.GetMesh ();
-                    return true;
-                }
                 return false;
             }
-            // check if mesh is shared between 2 objects that are not prefabs
-            else if (unityPrefabParent == null)
+            // export mesh as an instance if it is a duplicate mesh or a prefab
+            else if (unityGoMesh && SharedMeshes.ContainsKey(unityGoMesh))
             {
-                // check if same mesh has already been exported
-                MeshFilter unityGoMesh = unityGo.GetComponent<MeshFilter>();
-                if (unityGoMesh != null && MeshToFbxNodeMap.ContainsKey(unityGoMesh.sharedMesh))
+                if (Verbose)
                 {
-                    fbxMesh = MeshToFbxNodeMap[unityGoMesh.sharedMesh].GetMesh();
+                    Debug.Log (string.Format ("exporting instance {0}", unityGo.name));
                 }
-                // export mesh as normal and add it to list
-                else
+
+                if (SharedMeshes.TryGetValue(unityGoMesh, out FbxNode node))
```

Comment:
if you do this TryGetValue at the if statement on line 1406, you won't need to get the node here again if it is contained in the dict